### PR TITLE
Feature/phpstan ci

### PR DESCRIPTION
**Pull request by aragon999**
Add a PHPStan Github action job to statically analyze the plugin code and fix some minor PHPStan issues. 

Currently the service container xml is not built, but as far as I can see, this should be no problem for this plugin. This pull request should probably also only be merged if there is a release of the action. 

I only open this pull request to get feedback on the action and the Dockerimage. Do you think there is something missing?

To run it locally one can use the following command (inside the plugin directory):
```bash
docker run --rm -v $(pwd):/app aragon999/phpstan-shopware:v6.2.2 analyze .
```

See for a failed run here:
https://github.com/aragon999/FroshPlatformThumbnailProcessor/actions/runs/157919062
And a successful one:
https://github.com/aragon999/FroshPlatformThumbnailProcessor/actions/runs/158170012

For now I left the file `bin/phpstan-config-generator.php` which can probably removed. If not, this pull request needs to be adjusted, such that the replacement is working with the updated `phpstan.neon.dist`.

### CODE REVIEW

File: phpstan.neon.dist [hunk 1]
```diff
@@ -2,10 +2,12 @@ parameters:
     level: 7
     reportUnmatchedIgnoredErrors: false
     inferPrivatePropertyTypeFromConstructor: true
+    checkMissingIterableValueType: false
     symfony:
         constant_hassers: false
-        # the placeholder "%ShopwareHashedCacheDir%" will be replaced on execution by bin/phpstan-config-generator.php script
```
Comment (mitelg) [inline phpstan.neon.dist:7]: why did you remove this? 

if it is not needed anymore, the `phpstan-config-generator.php` is also obsolete :wink:

Comment (aragon999) [reply phpstan.neon.dist:7]: Yes, I think this can be removed (see also the comment in the pull request itself), since all Shopware files should be properly autoloaded as they are installed via composer in the Docker image. I would then also remove this file.

I am currently not sure if the container is still needed for the phpstan analysis or if this is obsolete, see also the following issue:
https://github.com/aragon999/phpstan-shopware-docker/issues/1

Comment (aragon999) [reply phpstan.neon.dist:7]: https://github.com/FriendsOfShopware/FroshPlatformThumbnailProcessor/pull/4/commits/4a5e872be95f876310c079079c9f14c0531b2292 :-)
